metabase/metabase
# Add an 'Exclude' option to 'Filter widget type'

Closes https://github.com/metabase/metabase/issues/30764

### Description


Adds an 'Exclude' option to 'Filter widget type':

<div float="left">
<img width="370" alt="Screenshot 2024-01-17 at 10 30 56 AM" src="https://github.com/kienstra/metabase/assets/4063887/a89fd49c-9230-4bde-a7c1-81b880aefb9a">
<img width="387" src="https://github.com/kienstra/metabase/assets/4063887/bd60fc63-f43b-4c0d-9ccd-55546000228e">
</div>

…so native queries can exclude by date, without having to select 'Date Filter' -> 'Exclude'

### How to verify

1. Create a new question
2. Click 'Sample Database'
3. Click the 'Accounts' table
4. Click 'View the SQL' 
5. Click 'Convert this question to SQL':

<img width="651" alt="Screenshot 2024-01-17 at 9 35 43 AM" src="https://github.com/kienstra/metabase/assets/4063887/1f8bee31-f070-4041-b0b0-ba615a717e3e">

6. Replace the SQL with this:

```sql
SELECT *
FROM accounts
WHERE {{dateOfCreation}}
```

7. From 'Variable type',  select 'Field Filter':
8. From 'Field to map to', select 'Accounts' -> 'Created At' (or any other date field)
9. From 'Filter widget type', select 'Exclude' (the new option in this PR):

<img width="996" alt="Screenshot 2024-01-17 at 9 40 46 AM" src="https://github.com/kienstra/metabase/assets/4063887/194999a9-8888-4012-8722-0e3e9dd3fb05">

10. Exclude September
11. Click 'Run query'
12. Expected: September doesn't show in `'CREATED_AT'` anymore:

https://github.com/kienstra/metabase/assets/4063887/3a4b7ba1-d968-443f-840a-9f7aae0d9473

(No audio in either video)

### Demo

https://github.com/metabase/metabase/assets/4063887/d5b19f0a-43fc-41e0-a883-c3cf28da7da9

### Checklist
